timbermill-local/src/main/java/com/datorama/oss/timbermill/cron/OrphansAdoptionJob.java

```diff
@@ -0,0 +1,30 @@
+package com.datorama.oss.timbermill.cron;
+
+import java.util.concurrent.atomic.AtomicBoolean;
+
+import org.quartz.Job;
+import org.quartz.JobExecutionContext;
+import org.slf4j.Logger;
+import org.slf4j.LoggerFactory;
+
+import com.datorama.oss.timbermill.ElasticsearchClient;
+import com.datorama.oss.timbermill.TaskIndexer;
+import com.datorama.oss.timbermill.common.ElasticsearchUtil;
+
+public class OrphansAdoptionJob implements Job {
+	private static final Logger LOG = LoggerFactory.getLogger(OrphansAdoptionJob.class);
+	private static final AtomicBoolean currentlyRunning = new AtomicBoolean(false);
+
+	@Override public void execute(JobExecutionContext context) {
+
+		if (currentlyRunning.compareAndSet(false, true)) {
```

Comment:
consider using @DisallowConcurrentExecution instead

Reply:
nice. I wasn't familiar with it